code/game/objects/items/devices/megaphone.dm

```diff
@@ -0,0 +1,67 @@
+/obj/item/device/megaphone
+	name = "megaphone"
+	desc = "A device used to project your voice. Loudly."
+	icon_state = "megaphone"
+	item_state = "radio"
+	w_class = 2
+	flags = FPRINT
+	siemens_coefficient = 1
+
+	var/spamcheck = 0
+	var/emagged = 0
+	var/insults = 0
+	var/list/insultmsg = list("FUCK EVERYONE!", "DEATH TO LIZARDS!", "ALL SECURITY TO SHOOT ME ON SIGHT!", "I HAVE A BOMB!", "CAPTAIN IS A COMDOM!", "FOR THE SYNDICATE!", "VIVA!", "HONK!")
+
+/obj/item/device/megaphone/attack_self(mob/living/carbon/human/user as mob)
+	if(user.client)
+		if(user.client.prefs.muted & MUTE_IC)
+			src << "<span class='warning'>You cannot speak in IC (muted).</span>"
+			return
+	if(!ishuman(user))
+		user << "<span class='warning'>You don't know how to use this!</span>"
+		return
+	if(ishuman(user) && (user.mind.miming)) //Humans get their muteness checked
+		user << "<span class='warning'>You find yourself unable to speak at all.</span>"
+		return
+	if(spamcheck)
+		user << "<span class='warning'>\The [src] needs to recharge!</span>"
+		return
+
+	var/message = copytext(sanitize(input(user, "Shout a message?", "Megaphone", null)  as text),1,MAX_MESSAGE_LEN)
+	if(!message)
+		return
+	message = capitalize(message)
+	if ((src.loc == user && usr.stat == 0))
+		if(emagged)
+			if(insults)
+				for(var/mob/O in (viewers(user)))
+					O.show_message("<B>[user]</B> broadcasts, <FONT size=3>\"[pick(insultmsg)]\"</FONT>",2) // 2 stands for hearable message
```

Comment:
miauw's sarcastic messages make me want to cry, it's like he doesn't actually want me to change anything.